gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -145,18 +138,32 @@ impl ClnRpcService {
             let socket = PathBuf::from(config.lightning_dir).join(config.rpc_file);
             let client = ClnRpc::new(socket).await.expect("connect to ln_socket");
 
-            let addr = match plugin.option("addr") {
-                Some(options::Value::String(address)) => SocketAddr::from_str(&address)
-                    .map_err(|e| ClnExtensionError::Error(anyhow!("{}", e)))?,
-                _ => unreachable!(),
+            // Parse configurations or read from
+            let listen: SocketAddr = match ClnExtensionOpts::try_parse() {
+                Ok(opts) => opts.listen,
+                // FIXME: cln_plugin doesn't yet support optional parameters
+                Err(_) => match plugin.option("listen") {
+                    Some(options::Value::String(listen)) => {
+                        if listen == "default-dont-use" {
+                            panic!(
```

Comment:
Eventually we might want to print an error and just exit via exit code instead of panic, but good enough for me for now.